platform-tests/src/test/java/org/junit/platform/commons/util/ClasspathScannerTests.java

```diff
@@ -341,4 +374,44 @@ private URI getTestClasspathRoot() throws Exception {
 		}
 	}
 
+	private static ClassLoader createClassLoaderReferencingJar(File jarFile) {
+		return new ClassLoader() {
+			@Override
+			public Enumeration<URL> getResources(final String name) throws IOException {
+				URL url = new URL(
+					"jar:file://" + jarFile.getAbsolutePath() + "!/BOOT-INF/classes!/org/junit/platform/commons");
+				return Collections.enumeration(Collections.singleton(url));
+			}
+		};
+	}
+
+	private static File createFakeSpringBootJar(final Class<?> clazz) throws IOException {
+		File fakeJarLocation = File.createTempFile("fakeSpringBoot", ".jar");
+		JarOutputStream target;
+		Manifest manifest = new Manifest();
+
+		manifest.getMainAttributes().put(Attributes.Name.MANIFEST_VERSION, "1.0");
+		target = new JarOutputStream(new FileOutputStream(fakeJarLocation), manifest);
+		fakeJarLocation.deleteOnExit();
```

Comment:
I only just noticed the conversation above about closing this issue. If it gets closed, then never mind. :)